### PR TITLE
guest_os_booting: add new case for ovmf firmware feature

### PR DESCRIPTION
**Pull request by meinaLi**
This PR automate:
VIRT-297067 - [OVMF] Boot vm with ovmf firmware feature

### CODE REVIEW

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --vt-machine-type q35 --test-runner=runner guest_os_booting.ovmf_firmware_feature
JOB ID     : 13a25e96257e71423599fdefb638b811ef95cf47
JOB LOG    : /var/lib/avocado/job-results/job-2023-06-16T03.21-13a25e9/job.log
 (1/4) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_firmware_feature.positive_test.enable_secure_boot: PASS (78.98 s)
 (2/4) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_firmware_feature.positive_test.disable_secure_boot: PASS (83.22 s)
 (3/4) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_firmware_feature.positive_test.not_support_secure_boot: PASS (63.33 s)
 (4/4) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_firmware_feature.negative_test.invalid_config: PASS (17.31 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-06-16T03.21-13a25e9/results.html
JOB TIME   : 244.53 s
```

**Comment by meinaLi** [comment]
Based on https://github.com/avocado-framework/avocado-vt/pull/3707

**Comment by meinaLi** [comment]
Label: modular_redesign

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_firmware_feature.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    try:
+        try:
+            guest_os.prepare_os_xml(vm_name, firmware_dict)
+        except xcepts.LibvirtXMLError as details:
+            if error_msg in str(details):
+                test.log.info("Can get expected error message: %s." % error_msg)
+            else:
+                test.fail("Failed to define the guest %s" % vm_name)
+        if not status_error:
```
Comment (chunfuwen) [inline libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_firmware_feature.py:40]: I doesn't find any place to start Vm operation, is this case intended?

Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_firmware_feature.py:40]: The function guest_os.check_vm_startup(vm, vm_name, error_msg) in line 41 can start vm.

**Review by cliping — APPROVED** [review]
LGTM

**Comment by meinaLi** [comment]
Add small changes based on the guest_os_booting_base.py based which is in https://github.com/autotest/tp-libvirt/pull/5025.

**Review by chunfuwen — APPROVED** [review]
lgtm

**Pull request merged**